src/SFA.DAS.AssessorService.Web/StartupConfiguration/AuthenticationStartup.cs

```diff
@@ -32,132 +31,228 @@ public static class AuthenticationStartup
+                                         Status = "Applying"
+                                     };
+
+                                     user = await CreateANewContact(newContact, contactClient, logger, signInId);
+                                 }
+
+                             }
+
+
+                             if (user != null)
+                             {
+                                 identity.AddClaim(new Claim("UserId", user?.Id.ToString()));
+                                 identity.AddClaim(new Claim(
+                                    "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/upn",
+                                    user?.Username));
+                                 if (user.EndPointAssessorOrganisationId != null)
+                                 {
+                                     var organisation =
+                                        await orgClient.GetEpaOrganisation(user.EndPointAssessorOrganisationId);
+
+                                     if (organisation.ApiEnabled && !string.IsNullOrEmpty(organisation.ApiUser))
+                                     {
+                                         identity.AddClaim(new Claim("http://schemas.portal.com/service",
+                                            Roles.ExternalApiAccess));
+                                         identity.AddClaim(new Claim("http://schemas.portal.com/service",
+                                            Roles.EpaoUser));
+                                     }
+
+                                     identity.AddClaim(new Claim("http://schemas.portal.com/ukprn",
+                                        organisation?.Ukprn == null ? "" : organisation?.Ukprn.ToString()));
+
+                                     var orgName = organisation.OrganisationData?.LegalName ??
+                                                  organisation.OrganisationData?.TradingName;
+
+                                     identity.AddClaim(new Claim("http://schemas.portal.com/orgname",
+                                        orgName));
+
+                                     identity.AddClaim(new Claim("http://schemas.portal.com/epaoid",
+                                        organisation?.OrganisationId));
+                                 }
+
+                                 identity.AddClaim(new Claim("display_name", user?.DisplayName));
+                                 identity.AddClaim(new Claim("email", user?.Email));
+
+                                 //Todo: Need to determine privileges dynamically
+                                 identity.AddClaim(new Claim("http://schemas.portal.com/service",
+                                     Privileges.ManageUsers));
+                             }
+                             
+                             context.Principal.AddIdentity(identity);
+                         }
+
+                     };
+                 });
```

Comment:
There is *a lot* going on in this Startup class now.  Can this be extracted someplace else to be unit tested?